### PR TITLE
Add developer set-up instructions for Internet Simulator.

### PR DESCRIPTION
**Pull request by islemaster**
Add a `CONTRIBUTING.md` specifically for NetSim which includes developer set-up instructions for Redis and Pusher.  Link to it from a couple of places so it's discoverable.  Remaining changes are my editor stripping trailing whitespace from the existing NetSim README.md.

### CODE REVIEW

File: apps/docs/netsim/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Contributing to Internet Simulator
+
+There is plenty of work to do on the Internet Simulator.  If you are internal to
+Code.org you can pick up a task from the [Internet Simulator Pivotal project](https://www.pivotaltracker.com/n/projects/1407902).  If not, please see
```
Comment (CaptainStack) [inline apps/docs/netsim/CONTRIBUTING.md]: Nit. Double space after period.

File: apps/docs/netsim/README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-The Internet Simulator is an app running on studio.code.org that simulates a 
-network for a group of users.  It does not actually establish peer-to-peer 
-connections between clients; instead, each client communicates with the 
-standard [app storage API](../../docs/client-api.md) on Code.org's servers, so 
-all clients interact with a common network state hosted with us.  Beyond that 
-network state, there isn't any intelligence on the server - clients are 
-trusted, and distribute work for running the simulation among themselves 
+The Internet Simulator is an app running on studio.code.org that simulates a
+network for a group of users.  It does not actually establish peer-to-peer
```
Comment (CaptainStack) [inline apps/docs/netsim/README.md]: Double space

File: apps/docs/netsim/README.md [hunk 3]
```diff
@@ -197,23 +202,23 @@ receiving the message.
...
-  3. The wires table comes next.  Any wire with an end connected to a missing 
+  3. The wires table comes next.  Any wire with an end connected to a missing
      node is invalid and may be cleaned up.
-  4. The messages table is next. Any message going to a missing node may be 
+  4. The messages table is next. Any message going to a missing node may be
      cleaned up.
-4. When it's done, the client releases the cleaning lock, letting another 
-   client take a turn.
\ No newline at end of file
+4. When it's done, the client releases the cleaning lock, letting another
```
Comment (CaptainStack) [inline apps/docs/netsim/README.md]: Should this be a 5?

**Comment by CaptainStack** [comment]
A few small comments. Other than that looks good.

File: apps/docs/netsim/CONTRIBUTING.md (hunk 1 above)
Comment (islemaster) [inline apps/docs/netsim/CONTRIBUTING.md]: Old habits die hard, man.  Fixing...

File: apps/docs/netsim/README.md (hunk 3 above)
Comment (islemaster) [inline apps/docs/netsim/README.md]: No.  Note the indentation; this is item #4 in the root-level list, not item #5 in the sub-list under item #3.

Now that you point this out though, I realize this whole section on shard cleaning is outdated and no longer applies.  I think I'll tear it out and replace it with something more current.

**Pull request merged**